### ISSUE TITLE
Documentation not using height that is available when the screen is split

### ISSUE THREAD

**Issue by David-Else** [hrsh7th/nvim-cmp#600]
```
NVIM v0.6.0
```
Using the Deno LS, Linux, Gnome Terminal

**Describe the bug**

The document height is too small when the screen is split vertically, there is space to display all of it, but I still need to scroll.

This is with a vertical split, it could fit on with no scrolling needed:
![Screenshot from 2021-12-01 21-56-37](https://user-images.githubusercontent.com/12832280/144320649-36f85f3a-9900-4591-8aad-db1e69d0dcf7.png)

This is without the split, it all fits on:
![Screenshot from 2021-12-01 21-56-07](https://user-images.githubusercontent.com/12832280/144320662-250778a8-b1d6-440e-b686-a386b2c8c1b8.png)

**Comment by hrsh7th** [comment]
It seems to be a neovim core's bug.

I think you can reproduce similar problem with `textDocument/hover` too.

**Comment by David-Else** [comment]
`textDocument/hover` works OK:

![Screenshot from 2021-12-02 09-41-40](https://user-images.githubusercontent.com/12832280/144397662-cdd63a89-0a4d-4568-aa2d-141cafa5ed64.png)
![Screenshot from 2021-12-02 09-42-01](https://user-images.githubusercontent.com/12832280/144397668-6b8db0ff-b84e-491b-9b7c-292c8ac3b3e9.png)

**Comment by hrsh7th** [comment]
No. What I want to say is that neovim's core API itself has been affected by split windows strangely.
_google translated_

https://user-images.githubusercontent.com/629908/144409951-1db32b2f-7a0e-4602-a669-eb5eeb284f28.mp4

**Comment by David-Else** [comment]
OK, thanks for the explanation. 

It would be great if someone could create an issue, I don't think I have the technical knowledge of Neovim to do that well.

**Comment by Iron-E** [comment]
This should be fixed by #472.
